lib/spark_api/models/saved_search.rb

```diff
@@ -69,30 +66,32 @@ def listings(args = {})
       end
 
       def newsfeeds
-        if @newsfeeds.nil?
-          response = SparkApi.client.get("/savedsearches/#{@attributes["Id"]}", :_expand => "NewsFeeds").first["NewsFeeds"]
-              # the response from the api is just a bunch of hashes, but we can turn them into Newsfeed instances
-          @newsfeeds = response.map { |hash| Newsfeed.new(hash) }
-        end
-        @newsfeeds
+        Newsfeed.collect(connection.get("#{self.class.path}/#{@attributes["Id"]}", 
+          :_expand => "NewsFeeds").first["NewsFeeds"])
       end
 
       def provided_search?
         false
       end
 
+      def self.qualifying_fields_for_newsfeed
+        # Newsfeed restriction criteria for saved searches:
+        # http://alpha.sparkplatform.com/docs/api_services/newsfeed/restrictions#criteria
+        %w(BathsTotal BedsTotal City CountyOrParish ListPrice Location MlsStatus PostalCode PropertyType 
```

Comment:
Not a big deal, but these could probably go in a class-level constant so we're not re-allocating the array+strings every time the method gets called